package/MDAnalysis/core/topologyattrs.py

```diff
@@ -1729,7 +1729,99 @@ def radius_of_gyration(group, wrap=False, **kwargs):
     @warn_if_not_unique
     @_pbc_to_wrap
     @check_atomgroup_not_empty
-    def shape_parameter(group, wrap=False):
+    def gyration_moments(group, wrap=False, unwrap=None, compound='group'):
+        r"""Moments of the gyration tensor.
+
+        The moments are defined as the eigenvalues of the gyration
+        tensor.
+
+        .. math::
+        
+            \mathsf{T} = \frac{1}{N} \sum_{i=1}^{N} (\mathsf{r_{i}} - 
+                \mathsf{r_{COM}})(\mathsf{r_{i}} - \mathsf{r_{COM}})
+
+        Where :math:`r_{COM}` is the center of mass.
+
+        See [Dima2004a]_ for background information.
+
+        Parameters
+        ----------
+        wrap : bool, optional
+            If ``True``, move all atoms within the primary unit cell before
+            calculation. [``False``]
+        unwrap : bool, optional
+            If ``True``, compounds will be unwrapped before computing their centers.
+        compound : {'group', 'segments', 'residues', 'molecules', 'fragments'}, optional
+            Which type of component to keep together during unwrapping.
+
+        Returns
+        -------
+        principle_moments_of_gyration : numpy.ndarray
+            Gyration vector(s) of (compounds of) the group in :math:`Å^2`.
+            If `compound` was set to ``'group'``, the output will be a single
+            value. Otherwise, the output will be a 1d array of shape ``(n,3)``
+            where ``n`` is the number of compounds.
+
+        .. versionadded:: 2.5.0
+        """
+
+        def __gyration(recenteredpos, masses):
```

Comment:
I would leave out the underscores and just call it `gyration()` or if you really want to highlight the private character then just a single underscore `_gyration()`. But double underscores are almost never needed and only make debugging harder.